[ILM] Convert node allocation component to TS and use hooks

yuliacech opened the pull request:
## Summary

This PR is a part of an ongoing effort to convert ILM codebase from JavaScript to TypeScript. In this PR, NodeAllocation component of the ILM policy edit page (in warm and cold phases) has been refactored. As a result, the component will not use the redux store for loaded node attributes anymore, but it will use a hook for that. Also a callout was added for the case, when the request fails. 

### Error loading node attributes 

#### Before 
- Loading indicator never disappears.
- No way to try reloading node attributes.
![node_attrs_error_before](https://user-images.githubusercontent.com/6585477/88198880-77ebb580-cc44-11ea-8493-93844d81710e.gif)

#### After 
- Loading indicator is only shown while the request is in progress.
- A callout was added where the user can try to reload node attributes.
![Error_loading_nodes_attributes](https://user-images.githubusercontent.com/6585477/88928810-8a37a600-d279-11ea-8219-acae97e988e5.gif)



### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/)
* Commit: 694b54c8198f9ce74466c47cfa1c74b491027c35
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/execution/node/306/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/","number":"63649","commit":"694b54c8198f9ce74466c47cfa1c74b491027c35"}]}
PIPELINE-->

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/)
* Commit: b3d49799b999ab5693b4cb20eb5d6d75ce80eff9
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/execution/node/316/log)

### History
* :broken_heart: [Build #63649](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/) failed 694b54c8198f9ce74466c47cfa1c74b491027c35

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/","number":"63649","commit":"694b54c8198f9ce74466c47cfa1c74b491027c35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/","number":"63917","commit":"b3d49799b999ab5693b4cb20eb5d6d75ce80eff9"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/)
* Commit: 4486489faff92ca0c2c0df1e92328dece0cf87c2

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/4486489faff92ca0c2c0df1e92328dece0cf87c2) | diff | [baseline](https://github.com/elastic/kibana/commit/52f3cc311d0a73792e2b38a21ea894f60aaa867e) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 266.8KB | -6.6KB | 273.5KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/4486489faff92ca0c2c0df1e92328dece0cf87c2) | diff | [baseline](https://github.com/elastic/kibana/commit/52f3cc311d0a73792e2b38a21ea894f60aaa867e) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 237.9KB | -375.0B | 238.2KB |

### History
* :broken_heart: [Build #63917](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/) failed b3d49799b999ab5693b4cb20eb5d6d75ce80eff9
* :broken_heart: [Build #63649](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/) failed 694b54c8198f9ce74466c47cfa1c74b491027c35

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/","number":"63649","commit":"694b54c8198f9ce74466c47cfa1c74b491027c35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/","number":"63917","commit":"b3d49799b999ab5693b4cb20eb5d6d75ce80eff9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/","number":"63982","commit":"4486489faff92ca0c2c0df1e92328dece0cf87c2"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/)
* Commit: af77efad2da539c0bfc811dd1e6798bc8f3c974b

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/af77efad2da539c0bfc811dd1e6798bc8f3c974b) | diff | [baseline](https://github.com/elastic/kibana/commit/1c791f39dac906f3a46b3703a82ba33e8f263a4b) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 266.8KB | -6.6KB | 273.5KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/af77efad2da539c0bfc811dd1e6798bc8f3c974b) | diff | [baseline](https://github.com/elastic/kibana/commit/1c791f39dac906f3a46b3703a82ba33e8f263a4b) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 237.9KB | -375.0B | 238.2KB |

### History
* :green_heart: [Build #63982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/) succeeded 4486489faff92ca0c2c0df1e92328dece0cf87c2
* :broken_heart: [Build #63917](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/) failed b3d49799b999ab5693b4cb20eb5d6d75ce80eff9
* :broken_heart: [Build #63649](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/) failed 694b54c8198f9ce74466c47cfa1c74b491027c35

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/","number":"63649","commit":"694b54c8198f9ce74466c47cfa1c74b491027c35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/","number":"63917","commit":"b3d49799b999ab5693b4cb20eb5d6d75ce80eff9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/","number":"63982","commit":"4486489faff92ca0c2c0df1e92328dece0cf87c2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/","number":"64631","commit":"af77efad2da539c0bfc811dd1e6798bc8f3c974b"}]}
PIPELINE-->

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/)
* Commit: 4cc958488cc802f28e0ca5cb11d1e34522b3f4a5

### Build metrics
#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/4cc958488cc802f28e0ca5cb11d1e34522b3f4a5) | diff | [baseline](https://github.com/elastic/kibana/commit/34e70ef8dd62e832428216d83943c441c8171c7f) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 267.9KB | -6.6KB | 274.5KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/4cc958488cc802f28e0ca5cb11d1e34522b3f4a5) | diff | [baseline](https://github.com/elastic/kibana/commit/34e70ef8dd62e832428216d83943c441c8171c7f) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 238.4KB | -375.0B | 238.8KB |

### History
* :green_heart: [Build #64631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/) succeeded af77efad2da539c0bfc811dd1e6798bc8f3c974b
* :green_heart: [Build #63982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/) succeeded 4486489faff92ca0c2c0df1e92328dece0cf87c2
* :broken_heart: [Build #63917](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/) failed b3d49799b999ab5693b4cb20eb5d6d75ce80eff9
* :broken_heart: [Build #63649](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/) failed 694b54c8198f9ce74466c47cfa1c74b491027c35

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/","number":"63649","commit":"694b54c8198f9ce74466c47cfa1c74b491027c35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/","number":"63917","commit":"b3d49799b999ab5693b4cb20eb5d6d75ce80eff9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/","number":"63982","commit":"4486489faff92ca0c2c0df1e92328dece0cf87c2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/","number":"64631","commit":"af77efad2da539c0bfc811dd1e6798bc8f3c974b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/","number":"65207","commit":"4cc958488cc802f28e0ca5cb11d1e34522b3f4a5"}]}
PIPELINE-->

yuliacech commented on x-pack/plugins/index_lifecycle_management/public/application/sections/components/learn_more_link.tsx:
```diff
@@ -0,0 +1,29 @@
+    <FormattedMessage id="xpack.indexLifecycleMgmt.learnMore" defaultMessage="Learn more" />
+  );
+  return (
+    <EuiLink href={createDocLink(docPath)} target="_blank">
```
I added an `external` prop and it creates an icon after the the link, but I still had to keep `target="_blank"` for it to open in a new tab. 
<img width="441" alt="Screenshot 2020-07-29 at 18 12 08" src="https://user-images.githubusercontent.com/6585477/88825536-fd80df80-d1c7-11ea-9e55-ec30905565c2.png">

yuliacech commented on x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/node_allocation/node_allocation.tsx:
```diff
@@ -0,0 +1,170 @@
+  nodeOptions.sort((a, b) => a.value.localeCompare(b.value));
+  if (nodeOptions.length) {
+    nodeOptions = [
+      { text: "Default allocation (don't use attributes)", value: '' },
```
yes, added `i18n.translate`

yuliacech commented on x-pack/plugins/index_lifecycle_management/__jest__/components/edit_policy.test.js:
```diff
@@ -37,9 +38,15 @@ import {
-initNotification({
-  addDanger: () => {},
-});
+initNotification(
```
yes, also added a mock to `UiMetrics`

yuliacech commented on x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/node_allocation/node_allocation.tsx:
```diff
@@ -0,0 +1,170 @@
+              defaultMessage="Unable to load node attributes."
+            />
+          }
+          color="warning"
```
I added the text (status code, status string, error message) extracted from the error object, similar to what the toast notification did before.

yuliacech commented:
Hi @alisonelizabeth , 
thank you for such a detailed code review! 
I implemented everything you commented on, could you please have another look?

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/)
* Commit: fc0633ecb10b927a4334d00695a3e2b0d03e79af
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/execution/node/316/log)

### History
* :green_heart: [Build #65207](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/) succeeded 4cc958488cc802f28e0ca5cb11d1e34522b3f4a5
* :green_heart: [Build #64631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/) succeeded af77efad2da539c0bfc811dd1e6798bc8f3c974b
* :green_heart: [Build #63982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/) succeeded 4486489faff92ca0c2c0df1e92328dece0cf87c2
* :broken_heart: [Build #63917](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/) failed b3d49799b999ab5693b4cb20eb5d6d75ce80eff9
* :broken_heart: [Build #63649](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/) failed 694b54c8198f9ce74466c47cfa1c74b491027c35

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63649/","number":"63649","commit":"694b54c8198f9ce74466c47cfa1c74b491027c35"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/","number":"63917","commit":"b3d49799b999ab5693b4cb20eb5d6d75ce80eff9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/","number":"63982","commit":"4486489faff92ca0c2c0df1e92328dece0cf87c2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/","number":"64631","commit":"af77efad2da539c0bfc811dd1e6798bc8f3c974b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/","number":"65207","commit":"4cc958488cc802f28e0ca5cb11d1e34522b3f4a5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/","number":"65260","commit":"fc0633ecb10b927a4334d00695a3e2b0d03e79af"}]}
PIPELINE-->

alisonelizabeth commented on x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/node_allocation/node_allocation.tsx:
```diff
@@ -0,0 +1,176 @@
+          defaultMessage="Learn about shard allocation"
+        />
+      }
+      docPath="shards-allocation.html"
```
I noticed while testing that this link is no longer valid. I think it should be: `modules-cluster.html#shard-allocation-awareness`

yuliacech commented:
Great suggestions, @alisonelizabeth ! 
I fixed the docs link and the "View list ..." link. There was also a type check error, so I added a type for the api error:  https://github.com/elastic/kibana/blob/1ccf12fac3e40449f61804d065e7d0c42b7f83a4/x-pack/plugins/index_lifecycle_management/public/application/services/http.ts#L45

yuliacech commented:
Hello @esdocs team, 

I added a new callout instead of a toast notification when there is an error loading node attributes. I used the same wording for error information but it seems a bit hard to read. So I was hoping for some feedback to improve it.

**Toast notification before:** 
![node_attrs_error_before](https://user-images.githubusercontent.com/6585477/88931468-29aa6800-d27d-11ea-9006-ce7cf15cc906.gif)

**New callout:** 
![Error_loading_nodes_attributes](https://user-images.githubusercontent.com/6585477/88931529-38911a80-d27d-11ea-8ce4-d44f981bcb70.gif)



The text is build like this: `{statusCode}: {errorString}. {message}` where `statusCode` is 404, 500 etc, `errorString` is "Bad Request", "Internal Server Error" etc and `message` is the information from API error but can be the same as `errorString`.
Some examples of current callout wordings: 
<img width="425" alt="Screenshot 2020-07-30 at 15 28 42" src="https://user-images.githubusercontent.com/6585477/88932429-5a3ed180-d27e-11ea-8091-7e5195eef86c.png">
<img width="421" alt="Screenshot 2020-07-30 at 15 36 31" src="https://user-images.githubusercontent.com/6585477/88932441-5f9c1c00-d27e-11ea-9413-37e8374fb062.png">
<img width="460" alt="Screenshot 2020-07-29 at 18 07 28" src="https://user-images.githubusercontent.com/6585477/88932467-688ced80-d27e-11ea-9c7a-40a5ca3b4c6e.png">

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/)
* Commit: 1ccf12fac3e40449f61804d065e7d0c42b7f83a4
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/execution/node/368/log)
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/execution/node/434/log)

### History
* :broken_heart: [Build #65260](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/) failed fc0633ecb10b927a4334d00695a3e2b0d03e79af
* :green_heart: [Build #65207](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/) succeeded 4cc958488cc802f28e0ca5cb11d1e34522b3f4a5
* :green_heart: [Build #64631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/) succeeded af77efad2da539c0bfc811dd1e6798bc8f3c974b
* :green_heart: [Build #63982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/) succeeded 4486489faff92ca0c2c0df1e92328dece0cf87c2
* :broken_heart: [Build #63917](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/) failed b3d49799b999ab5693b4cb20eb5d6d75ce80eff9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63917/","number":"63917","commit":"b3d49799b999ab5693b4cb20eb5d6d75ce80eff9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/","number":"63982","commit":"4486489faff92ca0c2c0df1e92328dece0cf87c2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/","number":"64631","commit":"af77efad2da539c0bfc811dd1e6798bc8f3c974b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/","number":"65207","commit":"4cc958488cc802f28e0ca5cb11d1e34522b3f4a5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/","number":"65260","commit":"fc0633ecb10b927a4334d00695a3e2b0d03e79af"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/","number":"65521","commit":"1ccf12fac3e40449f61804d065e7d0c42b7f83a4"}]}
PIPELINE-->

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/)
* Commit: a557a5bd4d3af61f73930690c1dcefc5d71424b7
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-intake](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/execution/node/308/log)

### History
* :broken_heart: [Build #65521](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/) failed 1ccf12fac3e40449f61804d065e7d0c42b7f83a4
* :broken_heart: [Build #65260](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/) failed fc0633ecb10b927a4334d00695a3e2b0d03e79af
* :green_heart: [Build #65207](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/) succeeded 4cc958488cc802f28e0ca5cb11d1e34522b3f4a5
* :green_heart: [Build #64631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/) succeeded af77efad2da539c0bfc811dd1e6798bc8f3c974b
* :green_heart: [Build #63982](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/) succeeded 4486489faff92ca0c2c0df1e92328dece0cf87c2

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/63982/","number":"63982","commit":"4486489faff92ca0c2c0df1e92328dece0cf87c2"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/","number":"64631","commit":"af77efad2da539c0bfc811dd1e6798bc8f3c974b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/","number":"65207","commit":"4cc958488cc802f28e0ca5cb11d1e34522b3f4a5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/","number":"65260","commit":"fc0633ecb10b927a4334d00695a3e2b0d03e79af"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/","number":"65521","commit":"1ccf12fac3e40449f61804d065e7d0c42b7f83a4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/","number":"65545","commit":"a557a5bd4d3af61f73930690c1dcefc5d71424b7"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/)
* Commit: 739be48412dbabf0acf824740dcd9d80e6829730

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/739be48412dbabf0acf824740dcd9d80e6829730) | diff | [baseline](https://github.com/elastic/kibana/commit/d8d830657b3c934c6a1b276683116b1f8b7e9bfd) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 177 | -1 | 178 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/739be48412dbabf0acf824740dcd9d80e6829730) | diff | [baseline](https://github.com/elastic/kibana/commit/d8d830657b3c934c6a1b276683116b1f8b7e9bfd) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 268.1KB | -6.4KB | 274.5KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/739be48412dbabf0acf824740dcd9d80e6829730) | diff | [baseline](https://github.com/elastic/kibana/commit/d8d830657b3c934c6a1b276683116b1f8b7e9bfd) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 238.4KB | -375.0B | 238.8KB |

### History
* :broken_heart: [Build #65545](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/) failed a557a5bd4d3af61f73930690c1dcefc5d71424b7
* :broken_heart: [Build #65521](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/) failed 1ccf12fac3e40449f61804d065e7d0c42b7f83a4
* :broken_heart: [Build #65260](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/) failed fc0633ecb10b927a4334d00695a3e2b0d03e79af
* :green_heart: [Build #65207](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/) succeeded 4cc958488cc802f28e0ca5cb11d1e34522b3f4a5
* :green_heart: [Build #64631](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/) succeeded af77efad2da539c0bfc811dd1e6798bc8f3c974b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/64631/","number":"64631","commit":"af77efad2da539c0bfc811dd1e6798bc8f3c974b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/","number":"65207","commit":"4cc958488cc802f28e0ca5cb11d1e34522b3f4a5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/","number":"65260","commit":"fc0633ecb10b927a4334d00695a3e2b0d03e79af"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/","number":"65521","commit":"1ccf12fac3e40449f61804d065e7d0c42b7f83a4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/","number":"65545","commit":"a557a5bd4d3af61f73930690c1dcefc5d71424b7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/","number":"65570","commit":"739be48412dbabf0acf824740dcd9d80e6829730"}]}
PIPELINE-->

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/)
* Commit: 062465a46d5452ff2154aa17160e883f0b5848e8

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/062465a46d5452ff2154aa17160e883f0b5848e8) | diff | [baseline](https://github.com/elastic/kibana/commit/278d3249b6331002b5d4a000e36764e807b64d08) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 177 | -1 | 178 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/062465a46d5452ff2154aa17160e883f0b5848e8) | diff | [baseline](https://github.com/elastic/kibana/commit/278d3249b6331002b5d4a000e36764e807b64d08) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 268.5KB | -6.4KB | 275.0KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/062465a46d5452ff2154aa17160e883f0b5848e8) | diff | [baseline](https://github.com/elastic/kibana/commit/278d3249b6331002b5d4a000e36764e807b64d08) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 238.6KB | -375.0B | 239.0KB |

### History
* :green_heart: [Build #65570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/) succeeded 739be48412dbabf0acf824740dcd9d80e6829730
* :broken_heart: [Build #65545](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/) failed a557a5bd4d3af61f73930690c1dcefc5d71424b7
* :broken_heart: [Build #65521](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/) failed 1ccf12fac3e40449f61804d065e7d0c42b7f83a4
* :broken_heart: [Build #65260](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/) failed fc0633ecb10b927a4334d00695a3e2b0d03e79af
* :green_heart: [Build #65207](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/) succeeded 4cc958488cc802f28e0ca5cb11d1e34522b3f4a5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65207/","number":"65207","commit":"4cc958488cc802f28e0ca5cb11d1e34522b3f4a5"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/","number":"65260","commit":"fc0633ecb10b927a4334d00695a3e2b0d03e79af"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/","number":"65521","commit":"1ccf12fac3e40449f61804d065e7d0c42b7f83a4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/","number":"65545","commit":"a557a5bd4d3af61f73930690c1dcefc5d71424b7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/","number":"65570","commit":"739be48412dbabf0acf824740dcd9d80e6829730"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/","number":"66059","commit":"062465a46d5452ff2154aa17160e883f0b5848e8"}]}
PIPELINE-->

lockewritesdocs commented:
@yuliacech, thanks for the ping 👍  I’m curious whether we want to display `{errorString}` if we’re essentially restating the same information in the `{message}`. I suggest refocusing the content to put the message first, removing `{errorString}`, and putting the `{statusCode}` last.

Pinging @probakowski and @leehinman to provide any additional context that might be useful to the user in understanding these messages.

**Unable to load node attributes**
Elasticsearch failed to load node attributes (400).

**Unable to load node attributes**
Elasticsearch cannot find node attributes (404).

**Unable to load node attributes**
Elasticsearch encountered an internal server error (500).

yuliacech commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66618/)
* Commit: 1f04f67f686284bf49116832f50e699b603975be

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/1f04f67f686284bf49116832f50e699b603975be) | diff | [baseline](https://github.com/elastic/kibana/commit/b26bd6175d11d52c577e62ebfc3186dea9d7ec58) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 177 | -1 | 178 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/1f04f67f686284bf49116832f50e699b603975be) | diff | [baseline](https://github.com/elastic/kibana/commit/b26bd6175d11d52c577e62ebfc3186dea9d7ec58) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 268.5KB | -6.4KB | 275.0KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/1f04f67f686284bf49116832f50e699b603975be) | diff | [baseline](https://github.com/elastic/kibana/commit/b26bd6175d11d52c577e62ebfc3186dea9d7ec58) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 238.3KB | -375.0B | 238.7KB |

### History
* :green_heart: [Build #66059](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/) succeeded 062465a46d5452ff2154aa17160e883f0b5848e8
* :green_heart: [Build #65570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/) succeeded 739be48412dbabf0acf824740dcd9d80e6829730
* :broken_heart: [Build #65545](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/) failed a557a5bd4d3af61f73930690c1dcefc5d71424b7
* :broken_heart: [Build #65521](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/) failed 1ccf12fac3e40449f61804d065e7d0c42b7f83a4
* :broken_heart: [Build #65260](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/) failed fc0633ecb10b927a4334d00695a3e2b0d03e79af

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65260/","number":"65260","commit":"fc0633ecb10b927a4334d00695a3e2b0d03e79af"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/","number":"65521","commit":"1ccf12fac3e40449f61804d065e7d0c42b7f83a4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/","number":"65545","commit":"a557a5bd4d3af61f73930690c1dcefc5d71424b7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/","number":"65570","commit":"739be48412dbabf0acf824740dcd9d80e6829730"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/","number":"66059","commit":"062465a46d5452ff2154aa17160e883f0b5848e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66618/","number":"66618","commit":"1f04f67f686284bf49116832f50e699b603975be"}]}
PIPELINE-->

yuliacech commented:
@elasticmachine merge upstream

yuliacech commented:
Thanks a lot for your suggestion, @lockewritesdocs ! 
I updated how error is displayed, would you mind having another look? 
<img width="421" alt="Screenshot 2020-08-06 at 15 39 44" src="https://user-images.githubusercontent.com/6585477/89539045-9680ad80-d7fb-11ea-8967-1e3a9ee9b81c.png">

lockewritesdocs commented on x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/node_allocation/node_allocation.tsx:
```diff
@@ -0,0 +1,177 @@
+          title={
+            <FormattedMessage
+              id="xpack.indexLifecycleMgmt.editPolicy.nodeAttributesLoadingFailedTitle"
+              defaultMessage="Unable to load node attributes."
```
Suggest removing the period here, but that's totally minor. If having a period at the end matches other Kibana error message conventions, then please ignore.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66893/)
* Commit: 594d01738fe86c8fc4775aa3a8e530c866c20b36

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/594d01738fe86c8fc4775aa3a8e530c866c20b36) | diff | [baseline](https://github.com/elastic/kibana/commit/bbfc4922d201bc26d7074035d2445dce36a5eeb7) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 177 | -1 | 178 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/594d01738fe86c8fc4775aa3a8e530c866c20b36) | diff | [baseline](https://github.com/elastic/kibana/commit/bbfc4922d201bc26d7074035d2445dce36a5eeb7) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 268.5KB | -6.4KB | 275.0KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/594d01738fe86c8fc4775aa3a8e530c866c20b36) | diff | [baseline](https://github.com/elastic/kibana/commit/bbfc4922d201bc26d7074035d2445dce36a5eeb7) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 238.3KB | -375.0B | 238.7KB |

### History
* :green_heart: [Build #66618](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66618/) succeeded 1f04f67f686284bf49116832f50e699b603975be
* :green_heart: [Build #66059](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/) succeeded 062465a46d5452ff2154aa17160e883f0b5848e8
* :green_heart: [Build #65570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/) succeeded 739be48412dbabf0acf824740dcd9d80e6829730
* :broken_heart: [Build #65545](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/) failed a557a5bd4d3af61f73930690c1dcefc5d71424b7
* :broken_heart: [Build #65521](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/) failed 1ccf12fac3e40449f61804d065e7d0c42b7f83a4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65521/","number":"65521","commit":"1ccf12fac3e40449f61804d065e7d0c42b7f83a4"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/","number":"65545","commit":"a557a5bd4d3af61f73930690c1dcefc5d71424b7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/","number":"65570","commit":"739be48412dbabf0acf824740dcd9d80e6829730"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/","number":"66059","commit":"062465a46d5452ff2154aa17160e883f0b5848e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66618/","number":"66618","commit":"1f04f67f686284bf49116832f50e699b603975be"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66893/","number":"66893","commit":"594d01738fe86c8fc4775aa3a8e530c866c20b36"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66916/)
* Commit: 3841a11e828f9d40988bac17776b641951ff1d99

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/3841a11e828f9d40988bac17776b641951ff1d99) | diff | [baseline](https://github.com/elastic/kibana/commit/bbfc4922d201bc26d7074035d2445dce36a5eeb7) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 177 | -1 | 178 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/3841a11e828f9d40988bac17776b641951ff1d99) | diff | [baseline](https://github.com/elastic/kibana/commit/bbfc4922d201bc26d7074035d2445dce36a5eeb7) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 268.5KB | -6.5KB | 275.0KB |

#### page load bundle size
| id | [value](https://github.com/elastic/kibana/commit/3841a11e828f9d40988bac17776b641951ff1d99) | diff | [baseline](https://github.com/elastic/kibana/commit/bbfc4922d201bc26d7074035d2445dce36a5eeb7) |
| --- | --- | --- | --- |
| indexLifecycleManagement | 238.3KB | -375.0B | 238.7KB |

### History
* :green_heart: [Build #66893](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66893/) succeeded 594d01738fe86c8fc4775aa3a8e530c866c20b36
* :green_heart: [Build #66618](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66618/) succeeded 1f04f67f686284bf49116832f50e699b603975be
* :green_heart: [Build #66059](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/) succeeded 062465a46d5452ff2154aa17160e883f0b5848e8
* :green_heart: [Build #65570](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/) succeeded 739be48412dbabf0acf824740dcd9d80e6829730
* :broken_heart: [Build #65545](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/) failed a557a5bd4d3af61f73930690c1dcefc5d71424b7

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65545/","number":"65545","commit":"a557a5bd4d3af61f73930690c1dcefc5d71424b7"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/65570/","number":"65570","commit":"739be48412dbabf0acf824740dcd9d80e6829730"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66059/","number":"66059","commit":"062465a46d5452ff2154aa17160e883f0b5848e8"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66618/","number":"66618","commit":"1f04f67f686284bf49116832f50e699b603975be"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66893/","number":"66893","commit":"594d01738fe86c8fc4775aa3a8e530c866c20b36"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/66916/","number":"66916","commit":"3841a11e828f9d40988bac17776b641951ff1d99"}]}
PIPELINE-->
